CMakeLists.txt

```diff
@@ -32,18 +32,20 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
+set(CMAKE_INSTALL_RPATH $<IF:$<PLATFORM_ID:Darwin>,"@loader_path/..","$ORIGIN">)
+set(CMAKE_INSTALL_REMOVE_ENVIRONMENT_RPATH YES)
+
 set(CMAKE_MSVC_RUNTIME_LIBRARY MultiThreadedDLL)
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_Swift_LANGUAGE_VERSION 6)
 set(CMAKE_Swift_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/swift)
 
-if(NOT SWIFT_SYSTEM_NAME)
-  if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
-    set(SWIFT_SYSTEM_NAME macosx)
-  else()
-    set(SWIFT_SYSTEM_NAME "$<LOWER_CASE:${CMAKE_SYSTEM_NAME}>")
-  endif()
-endif()
+option(SwiftTesting_INSTALL_NESTED_SUBDIR "Install libraries under a platform and architecture subdirectory" NO)
+set(SwiftTesting_INSTALL_LIBDIR "${CMAKE_INSTALL_LIBDIR}/swift$<$<NOT:$<BOOL:${BUILD_SHARED_LIBS}>>:_static>$<$<BOOL:${SwiftTesting_INSTALL_NESTED_SUBDIR}>:/${SwiftTesting_PLATFORM_SUBDIR}/${SwiftTesting_ARCH_SUBDIR}>")
```

Comment:
The existing logic which forms the library install path intentionally appends the path component `testing/` for Darwin toolchains:
```cmake
  if(APPLE)
    set(${result_var_name} "lib/${swift}/${swift_os}/testing" PARENT_SCOPE)
  else()
    set(${result_var_name} "lib/${swift}/${swift_os}" PARENT_SCOPE)
  endif()
```

Can we be sure that is preserved in this new logic? Several integrated tools rely on this.

Reply:
Okay, restored `/testing` suffix for Apple platforms when `Testing_INSTALL_NESTED_SUBDIR` is unset (default). If we want that to be `testing/macosx/arm64`, I'm happy to change that.